DependencyInjection/SonataMediaExtension.php

```diff
@@ -44,6 +44,27 @@ public function load(array $configs, ContainerBuilder $container)
         $loader->load('extra.xml');
         $loader->load('form.xml');
         $loader->load('gaufrette.xml');
+
+        $defS3 = $container->getDefinition('sonata.media.adapter.service.s3');
+        if (method_exists($defS3, 'setFactory')) {
+            // NEXT_MAJOR: Inline in gaufrette.xml
+            $defS3->setFactory(array('Aws\S3\S3Client', 'factory'));
+        } else {
+            // NEXT_MAJOR: Remove
+            $defS3->setFactoryClass('Aws\S3\S3Client');
+            $defS3->setFactoryMethod('factory');
+        }
+
+        $defOpenCloud = $container->getDefinition('sonata.media.adapter.filesystem.opencloud.objectstore');
```

Comment:
can we rename `$defOpenCloud`, this looks weird

Maybe `$openCloudDefinition` ??

/cc @greg0ire